pkg/component/plutono/plutono.go

```diff
@@ -84,6 +161,546 @@ func (p *plutono) WaitCleanup(ctx context.Context) error {
+				extensionsDashboards.WriteString(fmt.Sprintln(strings.ReplaceAll(strings.ReplaceAll(usersDashboard, "Grafana", "Plutono"), "loki", "vali")))
+			}
+		}
+
+		if extensionsDashboards.Len() > 0 {
+			if err := yaml.Unmarshal([]byte(extensionsDashboards.String()), &dashboards); err != nil {
+				return nil, err
+			}
+		}
+	}
+
+	for dashboardPath, dashboardEmbed := range requiredDashboards {
+		if err := fs.WalkDir(dashboardEmbed, dashboardPath, func(path string, dirEntry fs.DirEntry, err error) error {
+			if err != nil {
+				return err
+			}
+
+			normalizedPath := strings.TrimPrefix(strings.TrimPrefix(path, dashboardPath), "/")
+			if normalizedPath == "" {
+				// No need to process top level.
+				return nil
+			}
+
+			// Normalize to / since it will also work on Windows
+			normalizedPath = filepath.ToSlash(normalizedPath)
+
+			if dirEntry.IsDir() {
+				if len(sets.New[string](strings.Split(path, "/")...).Intersection(ignorePaths)) > 0 {
+					return filepath.SkipDir
+				}
+				return nil
+			}
+
+			data, err := dashboardEmbed.ReadFile(path)
+			if err != nil {
+				return fmt.Errorf("error reading %s: %s", normalizedPath, err)
+			}
+			dashboards[normalizedPath[strings.LastIndex(normalizedPath, "/")+1:]] = string(data)
+
+			return nil
+		}); err != nil {
+			return nil, err
+		}
+	}
+
+	// this is necessary to prevent hitting configmap size limit.
+	return convertToCompactJSON(dashboards)
+}
+
+func (p *plutono) getService() *corev1.Service {
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: p.namespace,
+			Labels:    getLabels(),
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name:     "web",
+					Port:     port,
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+			Selector: getLabels(),
+		},
+	}
+
+	if p.values.ClusterType == component.ClusterTypeSeed {
+		service.Labels = utils.MergeStringMaps(service.Labels, map[string]string{v1beta1constants.LabelRole: v1beta1constants.LabelMonitoring})
+	}
+
+	return service
+}
+
+func (p *plutono) getDeployment(providerConfigMapName, dataSourceConfigMapName, dashBoardConfigMapName string) *appsv1.Deployment {
+	sizeLimit := resource.MustParse("100Mi")
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: p.namespace,
+			Labels:    getLabels(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			RevisionHistoryLimit: pointer.Int32(1),
+			Replicas:             pointer.Int32(p.values.Replicas),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: getLabels(),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: utils.MergeStringMaps(getLabels(), map[string]string{
+						v1beta1constants.LabelNetworkPolicyToDNS:          v1beta1constants.LabelNetworkPolicyAllowed,
+						gardenerutils.NetworkPolicyLabel("logging", 3100): v1beta1constants.LabelNetworkPolicyAllowed,
```

Comment:
Introduce constant for https://github.com/gardener/gardener/blob/d30150fc0bdc5702aa316de265d0a576577fb9f4/pkg/component/logging/vali/vali.go#L448 and reuse it here? Same for the port, export it and use it here.